### ISSUE TITLE
Question: LabView compability broken after migrating from 4 to 5 because generics

### ISSUE THREAD

**Issue by BerndSprau** [fo-dicom/fo-dicom#1316]
Hi together, im not sure if this is the right method to ask, but i have a problem and hoping you can give me a hint how to solve it. 
We are using fo-dicom in a wrapped env and use it in LabView. The problem is, LabView can not support anything with generics in it, and DicomDataSet as well as DicomTags has now generics inside. 
`public sealed class DicomTag : IFormattable, IEquatable<DicomTag>, IComparable<DicomTag>, IComparable `
`public class DicomDataset : IEnumerable<DicomItem>, IEnumerable`

Before the migration, i coded wrappers for addOrUpdate and all functionality, but for DICOMTags this is not possible anymore. 

Is there any chance to generate DicomTags without the generics?

**Comment by amoerie** [comment]
So you're saying that your "special .NET environment" does not support generics? 
I'm afraid we won't be able to help you...

I also think it's unfair of you to say we "broke compatibility" while the old compatibility was purely accidental. You are running Fellow Oak DICOM on an entirely unsupported platform, and it was mostly a coincidence that you could get it to work via a workaround.

Don't get me wrong, if there was a very minor thing we could do and that would fix things for you, we would, but it doesn't sound so trivial unfortunately. Good luck.

**Comment by BerndSprau** [comment]
Ok yes, it was quite unfair to speak of "break compability", maybe i should have named it "break of evil workaroundable far from any supported usecase"? Sorry for that :) 
The Minor thing you can do maybe.. wold it be possible to have the two classes DICOMTags and DicomDataset without the generics? Or is my only changse to fork the projekt and edit it myselfe? (i would love to not have to do it to be honest)? Or remove the sealt tag, so i can subclass it? I dont know it, thats why i ask, maybe some of you can give me a hint?

**Comment by amoerie** [comment]
> wold it be possible to have the two classes DICOMTags and DicomDataset without the generics?

No, this would be a huge refactoring

> Or is my only changse to fork the projekt and edit it myselfe

I'm afraid it is. Then you will also see how much work this is. :-) 

> (i would love to not have to do it to be honest)

So do I.

> Or remove the sealt tag, so i can subclass it?

If we do so, it will just break in subtle ways. All of the existing code is written with the assumption DicomTag cannot be subclassed. 

>  I dont know it, thats why i ask, maybe some of you can give me a hint?

Generics have existed since .NET Framework 2.0. You are essentially asking us how Fellow Oak DICOM can be modified to run in .NET 1.0. My answer is it cannot without considerable work, which we won't do for you.

To be honest, my suggestion to you would be: either 
- A. figure out a way to make generics work in Labview, or 
- B. drop Labview, or 
- C. make a copy of fo-dicom and distill the things you need and strip those of generics. And then keep using that. If you want upgrades / improvements that we make here, you will need to repeat that process from time to time. That is the only way I see this working.

**Comment by gofal** [comment]
I also wanted to ask, if LabView uses such an old runtime, or if LabView only has this limitation in classes that are used as interface between labview and .net?
Because in second case, the Facade Design Pattern would be suitable. instead of forking fo-dicom and merging again and again, i would create some Models without interfaces, and then do lot of renundand code, to copy data beween the DicomDatasets and the Model. But this data-copying has to be implemented only once, and you still can be up-to-date with fo-dicom.

**Comment by BerndSprau** [comment]
@gofal You are right, LabView uses a Framework 4.6ish Framework under the hood. You cant call Methods or even see Classes, wich uses any type of Generics, but if the next Method has generics, this works just fine. 
Exactly this is what I've done, making a "LabViewFacade", and Generated my own DicomTags_cst, DicomUID_cst, DicomDataset (...) who just calls the Methods in the fo-dicom Librarys. 

Thanks alot.
